### DISCUSSION TITLE
custom docker-compose run commands

### DISCUSSION THREAD

**Discussion by bva-mike** [microsoft/vscode-docker#3064] [Q&A] [answered]
Hello, 

forgive me, but i couldn't find enough documentation around this:

i was wondering if there was a way to setup custom commands for docker-compose (https://docs.docker.com/compose/reference/run/) such that when i click on a docker-compose.yml file the command would appear in there?

![image](https://user-images.githubusercontent.com/51820538/125614122-89860238-ed72-4f71-8afa-c08cd3112507.png)

**Comment by karolz-ms** [comment]
@bva-mike I do not think there is a way to add custom commands to VS Code Explorer nodes other than by writing a VS Code extension. @philliphoff please chime in if I am missing something.

That said, you can use tasks to set up any Docker Compose invocations, either using [Docker Extension Compose task type](https://code.visualstudio.com/docs/containers/reference#_docker-compose-task), or just regular shell task. With a custom key binding like 
```json
    {
        "key": "alt+t",
        "command": "workbench.action.tasks.runTask"
    }
```
it can be quite easy to invoke these tasks. The limitation is, they won't be able to operate on any Compose files, only the ones configured for each task.

**Comment by philliphoff** [comment, marked as answer]
> @karolz-ms @bva-mike I do not think there is a way to add custom commands to VS Code Explorer nodes other than by writing a VS Code extension. @philliphoff please chime in if I am missing something.

Correct, to add commands to a file's context menu, you'd need to write an extension.  The good news is that writing extensions (especially for exposing individual commands) is fairly straightforward and the [documentation](https://code.visualstudio.com/api/get-started/your-first-extension) is good, and doesn't require you to actually publish it (if you don't want to).

**Comment by bva-mike** [comment]
@karolz-ms / @philliphoff - ah i see, thanks for the info guys! I'll take a look at writing a small extension!
